src/wallet_sync.rs

```diff
@@ -604,20 +786,25 @@ impl Wallet {
     pub fn find_incomplete_coinswaps(
         &self,
         rpc: &Client,
-    ) -> Result<HashMap<[u8; 20], Vec<(ListUnspentResultEntry, &WalletSwapCoin)>>, Error> {
+    ) -> Result<HashMap<[u8; 20], Vec<(ListUnspentResultEntry, &dyn SwapCoin)>>, Error> {
```

Comment:
Please change this whole function a little bit to return `HashMap::<[u8; 20]>, (Vec<(ListUnspentResultEntry, &IncomingSwapCoin)>, Vec<(ListUnspentResultEntry, &OutgoingSwapCoin)>)`. So make the IncomingSwapCoin and OutgoingSwapCoin be returned separately in different vectors.